### PR TITLE
feat(chrome): add chrome, example routes, and manage sidebar

### PR DESCRIPTION
**Pull request by seanforyou23**
This PR adds minimal support for routing, bootstrapping seed with Patternfly's standard Page component. I've stubbed out a 404-page route, and lazy loaded the support page - all for illustration sake.

Helps close https://github.com/patternfly/patternfly-react-seed/issues/52

Screen Shot 2019-07-02 at 9 51 09 PM

Screen Shot 2019-07-08 at 8 57 55 PM

### CODE REVIEW

File: src/app/NotFound/NotFound.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import { NavLink } from 'react-router-dom';
+import { Alert, PageSection } from '@patternfly/react-core';
+
+export function NotFound() {
+  return (
+    <PageSection>
+      <Alert variant="danger" title="404! This view hasn't been created yet." /><br />
+      <NavLink to="/" className="pf-c-nav__link">Back to Dashboard</NavLink>
```
Comment (cdcabrera) [inline src/app/NotFound/NotFound.tsx]: Question: So it looks like navigating back to Dashboard is missing something, probably an aspect of state? The reason, when you navigate back to `dashboard` its looking like the `pf-m-current` class isn't being applied to `dashboard` or removed from `bad route` in the left nav

  
Jul-03-2019 15-29-14

**Comment by cdcabrera** [comment]
An alternative approach, for the future, to take a look at... 

In some of the products we contribute towards we've been organizing the initial base component setup as 
```
      <PageLayout>
        <Router />
      </PageLayout>
```
Where Router actually has a separate `route types`, or constant, file in the form of an exported array. That might provide us with another possible iteration. Where we could separate out the current contents of `app` into a `pagelayout`, `router`, `routes`, and the actual component views loaded via routes. It does potential increase the verbosity, but something to consider.

File: src/app/index.tsx [hunk 2]
```diff
@@ -1,10 +1,21 @@
-import React, { Component } from 'react';
-import "@patternfly/react-core/dist/styles/base.css";
+import * as React from 'react';
+import '@patternfly/react-core/dist/styles/base.css';
```
Comment (dlabaj) [inline src/app/index.tsx]: Can we move the page code out of index.tsx and put it in there own components?  Beo yice to see it where there is a AppLayout which contains the Page AppToolbar for the main toolbar,AppHeader for logo, toolbar, etc, and AppSidebar for the vertical nav.

This way we keep index pretty small, and have smaller components to move around.

File: src/app/NotFound/NotFound.tsx (hunk 1 above)
Comment (seanforyou23) [reply src/app/NotFound/NotFound.tsx]: Should be fixed in the latest push

File: src/app/index.tsx (hunk 2 above)
Comment (seanforyou23) [reply src/app/index.tsx]: I broke the layout related code into its own component, didn't go so far as breaking out the header/sidebar as this gets into a grey area in terms of what we should recommend as a starting point, but what's there does generally follow the examples/demos so I'm feeling good about it. Let me know if you see any issues.

**Comment by seanforyou23** [comment]
Alrighty, made some updates - thanks for the feedbac! I'm gonna work on introducing code splitting for the app routes next, feel free to merge this if it's good to go, otherwise I'll lump those changes in with this PR as well.

**Pull request merged**